cpp/example_code/s3-crt/s3-crt-demo.cpp

```diff
@@ -0,0 +1,222 @@
+
+    if (outcome.IsSuccess()) {
+        std::cout << "Object content: " << outcome.GetResult().GetBody().rdbuf() << std::endl << std::endl;
+
+        return true;
+    }
+    else {
+        std::cout << "GetObject error:\n" << outcome.GetError() << std::endl << std::endl;
+
+        return false;
+    }
+}
+
+// Delete the Amazon S3 object from the bucket.
+bool DeleteObject(const Aws::S3Crt::S3CrtClient& s3CrtClient, const Aws::String& bucketName, const Aws::String& objectKey) {
+
+    std::cout << "Deleting object: \"" << objectKey << "\" from bucket: \"" << bucketName << "\" ..." << std::endl;
+
+    Aws::S3Crt::Model::DeleteObjectRequest request;
+    request.SetBucket(bucketName);
+    request.SetKey(objectKey);
+
+    Aws::S3Crt::Model::DeleteObjectOutcome outcome = s3CrtClient.DeleteObject(request);
+
+    if (outcome.IsSuccess()) {
+        std::cout << "Object deleted." << std::endl << std::endl;
+
+        return true;
+    }
+    else {
+        std::cout << "DeleteObject error:\n" << outcome.GetError() << std::endl << std::endl;
+
+        return false;
+    }
+}
+
+// 1. List all buckets under the account
+// 2. Create an Amazon S3 bucket
+// 3. Put an object to the bucket
+// 4. Get the object
+// 5. Delete the object
+// 6. Delete the bucket
+int main(int argc, char* argv[]) {
+
+    Aws::SDKOptions options;
+
+    // Override default log level for AWS common runtime libraries to prevent from being overwhelmed by logs from them.
+    options.loggingOptions.crt_logger_create_fn = []() {
+        return Aws::MakeShared<Aws::Utils::Logging::DefaultCRTLogSystem>(ALLOCATION_TAG, Aws::Utils::Logging::LogLevel::Warn);
+    };
+
+#if 0
+    // Override default global client bootstrap for AWS common runtime libraries here.
+    options.ioOptions.clientBootstrap_create_fn = []() {
+        Aws::Crt::Io::EventLoopGroup eventLoopGroup(0 /* cpuGroup */, 18 /* threadCount */);
+        Aws::Crt::Io::DefaultHostResolver defaultHostResolver(eventLoopGroup, 8 /* maxHosts */, 300 /* maxTTL */);
+        auto clientBootstrap = Aws::MakeShared<Aws::Crt::Io::ClientBootstrap>(ALLOCATION_TAG, eventLoopGroup, defaultHostResolver);
+        clientBootstrap->EnableBlockingShutdown();
+        return clientBootstrap;
+    };
+
+    // Override default global TLS connection options for AWS common runtime libraries here.
+    options.ioOptions.tlsConnectionOptions_create_fn = []() {
+        Aws::Crt::Io::TlsContextOptions tlsCtxOptions = Aws::Crt::Io::TlsContextOptions::InitDefaultClient();
+        Aws::Crt::Io::TlsContext tlsContext(tlsCtxOptions, Aws::Crt::Io::TlsMode::CLIENT);
+        return Aws::MakeShared<Aws::Crt::Io::TlsConnectionOptions>(ALLOCATION_TAG, tlsContext.NewConnectionOptions());
+    };
+#endif
+
+    Aws::InitAPI(options);
+    {
+        Aws::String bucket_name = "my-bucket";
+        Aws::String object_key = "my-object";
+        Aws::String region = Aws::Region::US_EAST_1;
+        double throughput_target_gbps = 5;
```

Comment:
I think they set the default bandwidth for CLI to be 25, might want to use that here too

Reply:
> I think they set the default bandwidth for CLI to be 25, might want to use that here too

Dengke saying that CLI is also 5Gbps

Reply:
I will keep 5 Gbps here.